docs/ADR-014-Cardano-API-WASM-library-for-browser.md

```diff
@@ -0,0 +1,167 @@
+
+# ADR-014: JavaScript API for Cardano via WASM-compiled cardano-api
+
+**Status:** Proposed
+
+**Date:** 2025-05-22
+
+## Context
+
+Decentralized Applications (DApps) are primarily utilized through web browsers. And Cardano DApps in Cardano would benefit from the extensive features provided by `cardano-api`.
+
+There already exist several JavaScript/TypeScript libraries and APIs that support DApps (e.g., Lucid Evolution, Helios, MeshJS, cardano-serialization-lib, cardano-sdk-js, Ogmios, TyphonJS), but porting `cardano-api` can offer advantages over other approaches:
+* **Comprehensiveness:** `cardano-api` is especially comprehensive in its functionalities, since it is developed in sync with `cardano-node`.
+* **Robustness:** Because `cardano-api` is developed in Haskell, it relies heavily in type correctness, and because it is heavily tested, it is especially trustworth. Because the JavaScript/TypeScript API would take advantage of the same code, the robustness would be inherited.
+* **Pure browser support:** DApp libraries tend to be more focused on supporting execution in the backend (through Node.js and Bun). There are good reasons for doing this, but explicitly supporting JavaScript in the browser has some advantages:
+  * **Incentivize adoption:** By making the library easier to try out, we would be reducing the friction to adoption.
+  * **Improved trust through decentralisation:** Since code is run in the browser, and thus the user's computer, it may be easier for the user to trust, even though transaction signing happens on the client side anyway.
```

Comment:
I think this point isn't super clear to me. I understand that tx building and signing is done on the client side anyway?

Reply:
Addressed here: https://github.com/input-output-hk/cardano-node-wiki/commit/f212dd5693050f1881c396cf99a1058d73c3f37d